### PR TITLE
fix: the emqx ctl command not working after relup

### PR DESCRIPTION
**Pull request by terry-xiaoyu**
Fixes https://emqx.atlassian.net/browse/EMQX-12935

Release version: v/e5.8.0

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: bin/emqx [hunk 1]
```diff
@@ -464,17 +474,40 @@ call_hocon() {
...
+    local rootdir="$1"
+    local nodename="$2"
+    check_emqx_process "${rootdir}" | $GREP -E "\s-s?name\s${nodename}" || true
+}
+
+do_find_emqx_process() {
+    local rootdir="$1"
     if [ -n "${EMQX_NODE__NAME:-}" ]; then
         # if node name is provided, filter by node name
         # shellcheck disable=SC2009
```
Comment (zmstone) [inline bin/emqx:495]: move to line 482

File: bin/emqx [hunk 2]
```diff
@@ -464,17 +474,40 @@ call_hocon() {
...
-        ps -ef | $GREP '[e]mqx' | $GREP -v -E '(remsh|nodetool)' | $GREP -E "\s-s?name\s${EMQX_NODE__NAME}" | $GREP -oE "\-[r]oot ${RUNNER_ROOT_DIR}.*" || true
+        check_emqx_process_of_node "${rootdir}" "${EMQX_NODE__NAME}"
     else
         # shellcheck disable=SC2009
```
Comment (zmstone) [inline bin/emqx:498]:
```suggestion
```

File: bin/emqx (hunk 1 above)
Comment (zmstone) [inline bin/emqx:495]:
```suggestion
```

File: bin/emqx [hunk 3]
```diff
@@ -464,17 +474,40 @@ call_hocon() {
         || die "call_hocon_failed: $*" $?
 }
 
-find_emqx_process() {
+check_emqx_process() {
+    local rootdir="$1"
     ## Find the running node from 'ps -ef'
     ##  * The grep args like '[e]mqx' but not 'emqx' is to avoid greping the grep command itself
     ##  * The running 'remsh' and 'nodetool' processes must be excluded
+    ps -ef | $GREP '[e]mqx' | $GREP -v -E '(remsh|nodetool)' | $GREP -oE "\-[r]oot ${rootdir}.*" || true
```
Comment (zmstone) [inline bin/emqx:482]:
```suggestion
    # shellcheck disable=SC2009
    ps -ef | $GREP '[e]mqx' | $GREP -v -E '(remsh|nodetool)' | $GREP -oE "\-[r]oot ${rootdir}.*" || true
```

Comment (zmstone) [reply bin/emqx:482]: shellcheck does not complain about this anymore.

File: bin/emqx (hunk 2 above)
Comment (terry-xiaoyu) [reply bin/emqx:498]: fixed

File: bin/emqx (hunk 1 above)
Comment (terry-xiaoyu) [reply bin/emqx:495]: fixed

**Review by zmstone — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-3651)

**Pull request merged**